dream-web/view/github.ml

```diff
@@ -0,0 +1,399 @@
+module Client = Ocaml_ci_api.Client
+module Common = Ocaml_ci_api.Common
+
+module Build_status = struct
+  include Client.Build_status
+
+  let class_name (t : t) =
+    match t with
+    | NotStarted -> "not-started"
+    | Failed -> "failed"
+    | Passed -> "passed"
+    | Pending -> "active"
+    | Undefined _ -> "undefined"
+end
+
+module StatusTree : sig
+  type key = string
+
+  type 'a tree = Leaf of 'a | Branch of key * 'a t
+  and 'a t = 'a tree list
+
+  val add : key list -> 'a -> 'a t -> 'a t
+end = struct
+  type key = string
+
+  type 'a tree = Leaf of 'a | Branch of key * 'a t
+  and 'a t = 'a tree list
+
+  let rec add k x ts =
+    match (k, ts) with
+    | [], ts -> ts @ [ Leaf x ]
+    | k :: ks, [] -> [ Branch (k, add ks x []) ]
+    | _ :: _, (Leaf _ as t) :: ts -> t :: add k x ts
+    | k :: ks, Branch (k', t) :: ts when Astring.String.equal k k' ->
+        Branch (k, add ks x t) :: ts
+    | _ :: _, (Branch _ as t) :: ts -> t :: add k x ts
+end
+
+let short_hash = Astring.String.with_range ~len:6
+
+let breadcrumbs steps page_title =
+  let open Tyxml.Html in
```

Comment:
Instead of opening `Tyxml.Html` every time, wouldn't it be better to open it once? Otherwise shouldn't the function by wrapped into a module and `Tyxml.Html` open at the top of this module?